### PR TITLE
add new API: optimizer.set_lr

### PR DESCRIPTION
**Pull request by zhwesky2010**
### PR types

New features
### PR changes

APIs
### Describe

New API：
```paddle.fluid.optimizer.set_lr(value)```

Setting learning rate manually in dygraph mode

### CODE REVIEW

File: python/paddle/fluid/optimizer.py [hunk 1]
```diff
@@ -296,6 +297,40 @@ def _create_global_learning_rate(self):
                 dtype='float32' if self._dtype is None else self._dtype,
                 persistable=True)
 
+    @framework.dygraph_only
+    def set_lr(self, value):
+        """
+        .. note::
+          **This API is ONLY available in Dygraph mode**
```
Comment (lanxianghit) [inline python/paddle/fluid/optimizer.py]: 改用新的写法
:api_attr: imperative

File: python/paddle/fluid/optimizer.py [hunk 2]
```diff
@@ -296,6 +297,40 @@ def _create_global_learning_rate(self):
...
+
```
Comment (lanxianghit) [inline python/paddle/fluid/optimizer.py]: 这个API似乎没有文档和代码示例？

Comment (zhwesky2010) [reply python/paddle/fluid/optimizer.py]: Done，已添加

File: python/paddle/fluid/optimizer.py (hunk 1 above)
Comment (zhwesky2010) [reply python/paddle/fluid/optimizer.py]: Done，已修改

**Review by lanxianghit — APPROVED** [review]
LGTM

**Review by saxon-zh — APPROVED** [review]
LGTM
